packages/material-ui-styles/src/index.d.ts

```diff
@@ -221,6 +221,7 @@ declare module '@material-ui/styles/useTheme' {
 declare module '@material-ui/styles/withStyles' {
   import * as React from 'react';
   import { Omit, PropInjector } from '@material-ui/core';
+  import { Theme } from '@material-ui/core';
```

Comment:
The `Theme` in `@material-ui/styles` is generic and doesn't necessarily implement the theme used in `@material-ui/core`. The typings need to make sure that the `defaultTheme` provided matches the one expected in the argument for `withStyles` (or `makeStyles`)

Reply:
what interface is more suitable instead of `Theme` from your PoV?

Reply:
Just a generic argument. We're completely agnostic to the shape.

Reply:
OK, will add a second generic argument, like so

```
export default function makeStyles<S extends Styles<any, any>, T = any>
```

will that work?

Reply:
Shouldn't the `T` also appear in `Styles` i.e. `<S extends Styles<T, any>, T = any>`.

Otherwise I could provide a `defaultTheme` that isn't understood by `styles`.